pyro/ops/welford.py

```diff
@@ -29,7 +29,7 @@ def update(self, sample):
         if self.diagonal:
             self._m2 += delta_pre * delta_post
         else:
-            self._m2 += torch.matmul(delta_post.unsqueeze(-1), delta_pre.unsqueeze(-2))
+            self._m2 += torch.ger(delta_post, delta_pre)
```

Comment:
Weird, I don't understand why there was a `matmul` at all, rather than the simpler
```py
delta_post.unsqueeze(-1) * delta_pre.unsqueeze(-2)
```

Reply:
The version that relies on broadcasting is a bit slower (since I guess it goes through the expand operation first) than matmul, but I doubt this was written with perf considerations in mind. 😄